### PR TITLE
Support for XPath functions in [dynamicxml] badge

### PR DESCRIPTION
**Pull request by platan**
Fixes https://github.com/badges/shields/issues/3202. 
XPath functions are well described at https://developer.mozilla.org/en-US/docs/Web/XPath/Functions. https://devhints.io/xpath is also a nice source of examples (with comparison to CSS selectors).

### CODE REVIEW

**Comment by calebcartwright** [comment]
Thanks @platan! I'll take a look at this in a few hours

**Comment by platan** [comment]
Badge reported in https://github.com/badges/shields/issues/3202 now works properly:

https://shields-staging-pr-4153.herokuapp.com/badge/dynamic/xml?label=version&query=string%28%2F%2Freleases%2Frelease%2F%40version%29&url=https%3A%2F%2Fraw.githubusercontent.com%2Fflathub%2Fio.github.Cockatrice.cockatrice%2Fmaster%2Fio.github.Cockatrice.appdata.xml

File: services/dynamic/dynamic-xml.service.js [hunk 1]
```diff
@@ -41,21 +41,25 @@ module.exports = class DynamicXml extends BaseService {
...
-        accum.push(node.firstChild.data)
-      }
+    if (
+      typeof values === 'string' ||
+      typeof values === 'number' ||
+      typeof values === 'boolean'
+    ) {
+      values = [values]
+    } else {
+      values = values.reduce((accum, node) => {
```
Comment (calebcartwright) [inline services/dynamic/dynamic-xml.service.js]: Are we sure that `xpath.select(...)` will always return a `string`, `number`, `boolean`, or an array? This seems like a superior fix for the type errors we were getting before, but I just want to make sure we can't still end up trying to call `.reduce` on an undefined `values` somehow

Comment (platan) [reply services/dynamic/dynamic-xml.service.js]: @calebcartwright  I wasn't able to find a query which returns something different than types mentioned above. But we can revert previous code to support these unusual values returned by `select`.

**Comment by platan** [comment]
@calebcartwright Thanks for a review and a good suggestion!

**Pull request merged**